client/landing/stepper/declarative-flow/flows/copy-site/copy-site.tsx

```diff
@@ -94,10 +98,72 @@ const copySite: Flow = {
 		const urlQueryParams = useQuery();
 		const sourceSiteSlug = urlQueryParams.get( 'sourceSlug' ) ?? '';
 		const sourceSite = useSite( sourceSiteSlug );
+		const {
+			setHideFreePlan,
+			setSignupDomainOrigin,
+			setDomainCartItem,
+			setSiteUrl,
+			setDomain,
+			setDomainCartItems,
+		} = useDispatch( ONBOARD_STORE ) as OnboardActions;
 
 		const submit = async ( providedDependencies: ProvidedDependencies = {} ) => {
 			switch ( _currentStepSlug ) {
 				case 'domains': {
+					if ( ! shouldRenderRewrittenDomainSearch() ) {
+						return navigate( 'create-site', {
+							sourceSlug: sourceSiteSlug,
+						} );
+					}
+
+					if ( providedDependencies.navigateToUseMyDomain ) {
+						const currentQueryArgs = getQueryArgs( window.location.href );
+						currentQueryArgs.step = 'domain-input';
+
+						let useMyDomainURL = addQueryArgs( '/use-my-domain', currentQueryArgs );
+
+						const lastQueryParam = providedDependencies.lastQuery as string | undefined;
+
+						if ( lastQueryParam !== undefined ) {
+							currentQueryArgs.initialQuery = lastQueryParam;
+							useMyDomainURL = addQueryArgs( useMyDomainURL, currentQueryArgs );
+						}
+
+						return navigate( useMyDomainURL as typeof _currentStepSlug );
+					}
+
+					setSiteUrl( providedDependencies.siteUrl as string );
+					setDomain( providedDependencies.suggestion as DomainSuggestion );
+					setDomainCartItem( providedDependencies.domainItem as MinimalRequestCartProduct );
+					setDomainCartItems( providedDependencies.domainCart as MinimalRequestCartProduct[] );
+					setSignupDomainOrigin( providedDependencies.signupDomainOrigin as string );
+
+					return navigate( 'create-site', {
+						sourceSlug: sourceSiteSlug,
+					} );
+				}
+
+				case 'use-my-domain': {
+					if (
+						providedDependencies &&
+						'mode' in providedDependencies &&
+						providedDependencies.mode &&
+						providedDependencies.domain
+					) {
+						const destination = addQueryArgs( '/use-my-domain', {
+							...getQueryArgs( window.location.href ),
+							step: providedDependencies.mode,
+							initialQuery: providedDependencies.domain,
+						} );
+						return navigate( destination as typeof _currentStepSlug );
+					}
+
+					if ( providedDependencies && 'domainCartItem' in providedDependencies ) {
```

Comment:
This will probably allow you to remove the casting below
```suggestion
					if ( providedDependencies && 'domainCartItem' in providedDependencies && domainCartItem ) {
```

Reply:
The type cast is still needed because `providedDependencies` is `unknown`. That's because the flow is still using the legacy architecture (v1) so it's not that smart.